resource/wrapper/retryresource/crud_resource.go

```diff
@@ -0,0 +1,216 @@
+package retryresource
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+
+	"github.com/giantswarm/operatorkit/resource/crud"
+)
+
+type crudResourceConfig struct {
+	BackOff backoff.Interface
+	CRUD    crud.Interface
+	Logger  micrologger.Logger
+}
+
+type crudResource struct {
+	backOff backoff.Interface
+	crud    crud.Interface
+	logger  micrologger.Logger
+}
+
+func newCRUDResource(config crudResourceConfig) (*crudResource, error) {
+	if config.BackOff == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.BackOff must not be empty", config)
+	}
+	if config.CRUD == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.CRUD must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	r := &crudResource{
+		backOff: config.BackOff,
+		crud:    config.CRUD,
+		logger:  config.Logger,
+	}
+
+	return r, nil
+}
+
+func (r *crudResource) Name() string {
+	return r.crud.Name()
+}
+
+func (r *crudResource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
+	var err error
+
+	var v interface{}
+	o := func() error {
+		v, err = r.crud.GetCurrentState(ctx, obj)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		return nil
+	}
+
+	n := func(err error, dur time.Duration) {
+		r.logger.LogCtx(ctx, "level", "warning", "message", "retrying due to error", "stack", fmt.Sprintf("%#v", err))
```

Comment:
When we are here:

```suggestion
		r.logger.LogCtx(ctx, "level", "warning", "message", "retrying due to error", "stack", microerror.Stack(err))
```

Reply:
I wanted to do this in this PR and noticed that the deps have to be updated. I guess because the unused file got purged by `dep`. I wanted to follow up on this in a separate PR.